packages/client/src/internal/conversions/modernToLegacyWhereClause.test.ts

```diff
@@ -943,4 +943,151 @@ describe(modernToLegacyWhereClause, () => {
       });
     });
   });
+
+  describe("RDP properties", () => {
+    const mockObjectType: ObjectOrInterfaceDefinition = {
+      type: "object",
+      apiName: "TestObject",
+    } as ObjectOrInterfaceDefinition;
+
+    it("should handle RDP properties at top level", () => {
+      const whereClause = {
+        department: "Engineering",
```

Comment:
The intellisense seems wrong. In my vscode I am doing:

```
      const whereClause: WhereClause<Employee> = {

        depart: 5,
        // department: "Engineering",
        reportCount: { $gte: 5 },
        managerName: "John",
      };
```

This is not erroring out.

And I am not seeing proper intellisense in regular client use either. I think the types are too permissive.